src/Serializer/JsonApiSerializer.php

```diff
@@ -586,6 +586,15 @@ private function serializeIncludedObjectsWithCacheKey($includeObjects, $linkedId
             if (!array_key_exists($cacheKey, $linkedIds)) {
                 $serializedData[] = $object;
                 $linkedIds[$cacheKey] = $object;
+            } else {
+                // Merge any missing relationships that may have been requested from alternative instances of this include
+                if (isset($object['relationships'])) {
+                    foreach ($object['relationships'] as $relationName => $relation) {
```

Comment:
Cool, thanks for the feedback :). Will try and find some time to put a quick test together & test out your suggestion (which looks at lot cleaner)